### PR TITLE
Dashboard [TAS-51]

### PR DESCRIPTION
**Pull request by femacedogomes**
Still need to work in filter by category and infinit scroll...

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -1,14 +1,106 @@
-import React from "react";
+"use client"
 
+import React, { useEffect, useState } from "react";
+import { SupplierCard } from '@/components/mols';
+import { CategoryCard } from '@/components/atoms'
+import SupLogo from '@/images/easytolive/logo/logotipo-fundoazulroxo.svg'
```
Comment (pedrohbl03) [inline src/app/page.tsx:6]: SupplierLogo. 

We make code for others. If u use generic therms the code can be so hard for reading by other people

File: src/app/page.tsx [hunk 2]
```diff
@@ -1,14 +1,106 @@
...
+import SearchCategory from '@/app/searchCategory'
+import supplierService from '@/service/supplier.service'
```
Comment (pedrohbl03) [inline src/app/page.tsx:8]: CamelCase -> SupplierService

File: src/app/page.tsx [hunk 3]
```diff
@@ -1,14 +1,106 @@
...
+
+  const getAllCategorys = async () => {
+    const res: any = await supplierService.getSupplierCategories();
+    return res;
+  }
+
+  useEffect(() => {
+    getAllCategorys()
+    .then((res)=> setCategorys(res?.data?.supplierCategories?.results))
+    .catch((error)=> console.log(error))
```
Comment (pedrohbl03) [inline src/app/page.tsx:32]: Toast working now. We can show a toats here? 

If we receive a error to get all Categories?

File: src/app/page.tsx [hunk 4]
```diff
@@ -1,14 +1,106 @@
...
 function PageHome() {
+  const [suppliers , setSuppliers] = useState([])
+  const [search, setSearch] = useState('');
+  const [textSearched] = useDebounce(search, 1000);
+  const [categorys, setCategorys] = useState([]);
+  const [pageNumber, setPageNumber] = useState(1);
+  const [hasMore, setHasMore] = useState(true);
+  const [supplierCategoryFilter, setSupplierCategoryFilter] = useState('');
+
+  const getAllCategorys = async () => {
```
Comment (pedrohbl03) [inline src/app/page.tsx:24]: Categories*

File: src/app/page.tsx [hunk 5]
```diff
@@ -1,14 +1,106 @@
...
+import { ISupplier, ISupplierList } from '@/types/supplier';
+import { useDebounce } from 'use-debounce';
+import { categorieProps } from '@/components/atoms/CategoryCard';
+import InfiniteScroll from 'react-infinite-scroll-component';
 
 function PageHome() {
+  const [suppliers , setSuppliers] = useState([])
+  const [search, setSearch] = useState('');
+  const [textSearched] = useDebounce(search, 1000);
+  const [categorys, setCategorys] = useState([]);
```
Comment (pedrohbl03) [inline src/app/page.tsx:19]: categories*

File: src/app/page.tsx [hunk 6]
```diff
@@ -1,14 +1,106 @@
...
+    setSuppliers( 
+      pageNumber === 1 
+        ? res?.data?.results 
+        : suppliers.concat(res?.data?.results)
+    );
+
+  useEffect(() => {
+    const data = {
+      page: pageNumber,
+      ...(textSearched ? { name: textSearched } : {}),
```
Comment (pedrohbl03) [inline src/app/page.tsx:63]:
```suggestion
      ...(textSearched && { name: textSearched }),
```

File: src/app/page.tsx [hunk 7]
```diff
@@ -1,14 +1,106 @@
...
+      ...(supplierCategoryFilter ? { supplierCategory: supplierCategoryFilter } : {}),
```
Comment (pedrohbl03) [inline src/app/page.tsx:64]:
```suggestion
      ...(supplierCategoryFilter && { supplierCategory: supplierCategoryFilter }),
```

File: src/components/mols/SupplierCard/index.tsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+import Image, { StaticImageData } from 'next/image'
+import ArrowRight from '@/images/easytolive/icons/arrow-next-right.svg'
+
+interface SupplierCardProps {
+  couponsAvaible: number;
+  supplierCategory: string
+  supplierImage: string | StaticImageData
+  name: string
+  avaliation?: string
+  onClick?: (e :any)=> void
```
Comment (pedrohbl03) [inline src/components/mols/SupplierCard/index.tsx:14]: Existe uma tipagem de evento para o click React.MouseEvent<HTMLButtonElement>

Comment (pedrohbl03) [inline src/components/mols/SupplierCard/index.tsx:14]:
```suggestion
  onClick?: (e: React.MouseEvent<HTMLDivElement>)=> void
```

**Review by pedrohbl03 — CHANGES_REQUESTED** [review]
Resolve the comment's! 

Good job so far

File: src/app/page.tsx (hunk 1 above)
Comment (femacedogomes) [reply src/app/page.tsx:6]: really nice point... i have to think more like that

**Review by pedrohbl03 — APPROVED** [review]

**Pull request merged**